asmcomp/riscv/emit.mlp

```diff
@@ -39,12 +39,17 @@ let frame_size env =
 let slot_offset env loc cls =
   match loc with
   | Local n ->
-      if cls = 0
-      then env.stack_offset + env.f.fun_num_stack_slots.(1) * size_float
-           + n * size_int
-      else env.stack_offset + n * size_float
-  | Incoming n -> frame_size env + n
-  | Outgoing n -> n
+      ("sp",
+       if cls = 0
+       then env.stack_offset + env.f.fun_num_stack_slots.(1) * size_float
+            + n * size_int
+       else env.stack_offset + n * size_float)
+  | Incoming n ->
+      ("sp", frame_size env + n)
+  | Outgoing n ->
+      ("sp", n)
+  | Domainstate n -> 
+      ("s11", n + Domainstate.(idx_of_field Domain_extra_params) * 8)
```

Comment:
The comment in `proc.ml` points to `"s0"` as the register holding the domain state pointer.

Reply:
That's not what I see: https://github.com/ocaml/ocaml/blob/4e9eb755e60d0f0f4e458eb3f7af26bce5d8e2ba/asmcomp/riscv/proc.ml#L44

Reply:
Sorry, I must have been looking at an older version (I'm working on too many versions at once...).